service/src/main/java/tds/config/repositories/impl/TimeLimitConfigurationRepositoryImpl.java

```diff
@@ -0,0 +1,104 @@
+package tds.config.repositories.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.IncorrectResultSizeDataAccessException;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.stereotype.Repository;
+import tds.config.TimeLimitConfiguration;
+import tds.config.repositories.TimeLimitConfigurationRepository;
+import tds.config.repositories.impl.mappers.TimeLimitsRowMapper;
+
+import javax.sql.DataSource;
+import java.util.Optional;
+
+@Repository
+public class TimeLimitConfigurationRepositoryImpl implements TimeLimitConfigurationRepository {
+    private static final Logger LOG = LoggerFactory.getLogger(ConfigRepositoryImpl.class);
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public TimeLimitConfigurationRepositoryImpl(DataSource dataSource) {
+        this.jdbcTemplate = new NamedParameterJdbcTemplate(dataSource);
+    }
+
+    @Override
+    public Optional<TimeLimitConfiguration> findTimeLimitConfiguration(final String clientName) {
+        final SqlParameterSource parameters = new MapSqlParameterSource("clientName", clientName);
+
+        final String SQL =
+                "SELECT \n" +
+                "   clientname, \n" +
+                "   _efk_testid AS assessmentId, \n" +
+                "   environment, \n" +
+                "   opprestart AS examRestartWindowMinutes, \n" +
+                "   oppdelay AS examDelayDays, \n" +
+                "   interfacetimeout AS interfaceTimeoutMinutes, \n" +
+                "   requestinterfacetimeout AS requestInterfaceTimeoutMinutes, \n" +
+                "   tacheckintime AS taCheckinTimeMinutes \n" +
+                "FROM \n" +
+                "   session.timelimits \n" +
+                "WHERE \n" +
+                "   clientname = :clientName \n" +
+                "   AND _efk_testid IS NULL";
+
+        Optional<TimeLimitConfiguration> maybeTimeLimitConfig;
+        try {
+            maybeTimeLimitConfig = Optional.of(
+                    jdbcTemplate.queryForObject(
+                            SQL,
+                            parameters,
+                            new TimeLimitsRowMapper()));
+        } catch(IncorrectResultSizeDataAccessException e) {
+            LOG.debug("{} did not return results for clientName = {}", SQL, clientName);
+            maybeTimeLimitConfig = Optional.empty();
+        }
+
+        return maybeTimeLimitConfig;
+    }
+
+    @Override
+    public Optional<TimeLimitConfiguration> findTimeLimitConfiguration(final String clientName, final String assessmentId) {
```

Comment:
it feels like if we added logic here that says if assessmentId == null, then `return findTimeLimitConfiguration(clientName)` we would allow the consumer to use either method.  It feels like this will start a debate but do we need both public methods?  

It's probably the .NET in me since I would make 1 method with an optional assessmentId parameter that defaults to null and calls a helper method since the SQL is different enough in the 2 implementations.  So i feel comfortable calling this with NULL as the second parameter myself as the caller.  But I have this feeling I might be alone on that :)